### PR TITLE
`diseasystores` for Individual level data

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR adds an example of a `diseasystore` built on individual level data.

This serves two goals:
1) To teach users how to utilise `diseasystore` in its most powerful form (namely with individual level data)

2) To have a `diseasystore` with individual level data in the testing pipeline (using synthetic data we can freely distribute)

> [!TIP]
> To review this PR, consider reading the new vignette first "extending-diseasystore-example" since it writes the thoughts behind `DiseasystoreSimulist` in clear text.
> You can also grab the rendered pkgdown output [here](https://github.com/ssi-dk/diseasystore/actions/runs/11680100831/artifacts/2145371215) (unzip, go to "articles" then open "extending-diseasystore-example.html")

### Approach
A synthetic "linelist" dataset was created using `simulist`.

A `diseasystore` was built surrounding this dataset (`DiseasystoreSimulist`). 

The design of this `diseasystore` is described in detail in the example vignette added in this PR.

With the individual level data of the new `DiseasystoreSimulist`, we use more date and age operations which are poorly implemented in SQLite and inconsistent in `dbplyrs` translations. Accounting for all of these particularities made this "pedagogical" example needlessly complicated. I could not simultaneously make the diseasystore a pedagogical example and support SQLite and SQL Server as backends. We can get SQL Server supported if we create a `add_days()` age helper akin to the `add_years()` age helper we already have (tracked in #183).  

### Known issues
This PR  builds on top of a lot of un-merged PRs that should be merged before this one is ready.
* [x] ~~#125~~
* [x] ~~#138~~
* [x] ~~#154~~
* [x] ~~#155~~
* [x] ~~#158~~
* [x] ~~#160~~
* [x] ~~#161~~
* [x] ~~#172~~
* [x] ~~#173~~
* [x] ~~#175~~
* [x] ~~#176~~
* [x] ~~#177~~

~~In addition, we need SCDB > 0.4 for the tests to work on all backends~~
* [x] ~~https://github.com/ssi-dk/SCDB/issues/150~~

* [x] ~~Age wanings on SQLite needs to be handled. Can we use  [pkgcond](https://cran.r-project.org/web/packages/pkgcond/index.html)?~~

This `diseasystore` does not support SQLite since the lack of date support was to great an obstacle to handle.
One of the goals of `DiseasystoreSimulist` was to be a pedagogical example of how to build a `diseasystore` with individual level data, and handling dates in SQLite made this impossible.

It also does not support `SQL Server` currently, as we need a new age helper `add_days()` to handle the remaining date operations. 

* [x]  ~~Check that we do not interfere with the new benchmarks due to the extra "ds" argument #144~~

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # diseasystore (development version)
 
-## New Features:
+## New Features
```
Comment (RasmusSkytte) [inline NEWS.md:3]:
```suggestion
## New Features:
```

File: vignettes/extending-diseasystore-example.Rmd [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+```
+
+Notice that we again create a new instance of `FeatureHandler` and assign it to the `simulist_sex` field in the
+private list.
+
+The computation of the feature is mostly straight forward, but we would like to use the validity period computed
+in the `birth` feature (the same logic applies for the validity period of the "sex" information as the "birth"
+information)
+
+Therefore, we immediately see a difference in the function signature of the `compute()` function, where we no include
```
Comment (SofiaOtero) [inline vignettes/extending-diseasystore-example.Rmd:229]: Therefore, we immediately see a difference in the function signature of the `compute()` function, where we don't include

**Comment by SofiaOtero** [comment]
Looks good with the new implementation 👍. Maybe an example on how you can specify the regex pattern `$observables_regex` to match the users own observables?

**Comment by RasmusSkytte** [comment]
> Looks good with the new implementation 👍. Maybe an example on how you can specify the regex pattern `$observables_regex` to match the users own observables?

Good idea. I have added an additional paragraph now

File: vignettes/extending-diseasystore-example.Rmd [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+      # Using birth date, compute the age at the start of the study period
+      age <- ds$get_feature("birth", start_date, end_date, slice_ts) |>
+        dplyr::mutate(
+          age_at_start = as.integer(
+            !!age_on_date("birth", start_date, conn = ds %.% target_conn)
+          )
+        ) |>
+        dplyr::compute()
+
+      # Now, compute the next birth dates of the individual
```
[the remark this answers is not in the document]
Comment (RasmusSkytte) [reply vignettes/extending-diseasystore-example.Rmd:274]: We compute, for each individual, all their birthdates that fall in the study period.

**Pull request merged**